src/main/java/de/cerus/jdasc/JDASlashCommands.java

```diff
@@ -214,20 +214,41 @@ private JDASlashCommands() {
      * @return The command id
      */
     public static CompletableFuture<Long> submitGlobalCommand(final ApplicationCommand command, final ApplicationCommandListener listener) {
-        return discordHttpClient.submitGlobalCommand(command).thenApply(response -> {
-            final JsonObject object;
-            try {
-                object = JsonParser.parseString(response.body().string()).getAsJsonObject();
-            } catch (final IOException e) {
-                throw new CompletionException(e);
+        try {
+            List<ApplicationCommand> globalCommands = getGlobalCommands().get();
```

Comment:
Have you tested this? I'm pretty sure this won't work. getGlobalCommands() returns you a future because the value is not yet present. Take a look at whenComplete()

Reply:
im using .get(), it completes the future. when complete is probs better tho